src/analysis/vector/geometry_checker/qgsgeometrycheckfactory.h

```diff
@@ -62,6 +62,27 @@ class QgsGeometryCheckFactoryT : public QgsGeometryCheckFactory
     {
       return new T( context, geometryCheckConfiguration );
     }
+
+    QString name() const override
+    {
+      return static_cast<T>( 0 ).errorDescription();
```

Comment:
well you're the computer guy ;)
The pointer was failiing with reinterpret_cast but I think I did not try with static_cast.

Reply:
I actually wonder how the cast to value succeeds for the SelfIntersection check. Casting to a `nullptr` to a pointer of a certain type on the other hand sounds like a safe thing to do (as long as you don't mess with data in the pointer).

Reply:
Fuck... of course.
One is not allowed to use any member variables of a `nullptr`. So the test for the geometry type in the parent class will crash for sure, this will have to be replaced with an implementation inside the method rather than passing it to the parent class' constructor. Not sure about `tr()`, but if it does, that could be replaced with `QCoreApplication.translate()`...

Reply:
we could make a pure virtual method compatibleTypes instead of using the member